de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/dataaccess/CustomJsonProvider.java

```diff
@@ -113,27 +113,78 @@ public Object readFrom(Class<Object> type, Type genericType, Annotation[] annota
 		AtomicExceptionReference<IOException> atomicIoException = new AtomicExceptionReference<>();
```

Comment:
Something is off with the exceptions. They are instantiated here, not passed to the readFromCommand and then the readFromCommand has its own exception data.

Maybe we can simplify this by
* Completely moving the exception objects into the ReadFromCommand
* Making a throwExceptionsIfSet() method in the ReadFromCommand that deals with throwing the indificual exceptions when needed

Reply:
You are right this got messed up, I will change it